Bootstrap.cs

```diff
@@ -76,28 +77,41 @@ private void GetAnswer(string clipboardText)
         {
             try
             {
-                foreach (IAction action in serviceProvider.GetServices<IAction>())
+                //foreach (IAction action in serviceProvider.GetServices<IAction>())
+                //{
+                //    // disconnect events from the clipboard.
+                //    clipboardMonitor.ClipboardChanged -= ClipboardMonitor_ClipboardChanged;
+                //    // run the action
+                //    ActionResult actionResult = action.TryExecute(clipboardText);
+                //    // re attach the event
+                //    clipboardMonitor.ClipboardChanged += ClipboardMonitor_ClipboardChanged;
+
+                //    // if this action processed the command, exit the loop.
+                //    if (actionResult.IsProcessed)
+                //    {
+                //        if (!String.IsNullOrWhiteSpace(actionResult.Title) || !String.IsNullOrWhiteSpace(actionResult.Description))
+                //        {
+                //            ShowNotification(actionResult);
+                //            Clipboard.Clear();
+                //        }
+                //        break;
+                //    }
+
+                //}
+
+                var service = serviceProvider.GetServices<IAction>().FirstOrDefault(s => s.Matches(clipboardText));

```

Comment:
Slick, only using the services (dependencies) needed. This will speed up the processes.